When I run a clean build locally, I run into this consistently while building (tested on a different machine as well)
```
C:\GitHub\aspnet\AspNetCore\src\Razor\Razor.Design\src\Microsoft.AspNetCore.Razor.Design.csproj(70,5): error MSB3027: Could not copy "C:\GitHub\aspne t\AspNetCore\src\Razor\Razor.Tasks\src\obj\Release\net46\Microsoft.AspNetCore.Razor.Tasks.dll" to "bin\Release\netstandard2.0\\tasks\net46\Microsoft. AspNetCore.Razor.Tasks.dll". Exceeded retry count of 10. Failed. The file is locked by: "MSBuild.exe (9944), MSBuild.exe (4680), MSBuild.exe (17352)"
```

On a second build everything passes including these tests. This is similar to what @NTaylorMullen is investigating with https://github.com/aspnet/AspNetCore-Internal/issues/1621. What is different here is that it doesn't seem to happen on the CI.

--- a/src/Razor/Razor.Design/src/Microsoft.AspNetCore.Razor.Design.csproj
+++ b/src/Razor/Razor.Design/src/Microsoft.AspNetCore.Razor.Design.csproj
@@ -35,7 +35,6 @@
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
 
   <PropertyGroup>
-    <GenerateNuspecDependsOn>_BuildDependencyProjects;$(GenerateNuspecDependsOn)</GenerateNuspecDependsOn>
     <BuildDependsOn>_BuildDependencyProjects;$(BuildDependsOn)</BuildDependsOn>
   </PropertyGroup>
 
@@ -92,7 +91,7 @@
     <Error Text="_RazorTool is empty. This is a bug" Condition="'@(_RazorTool)'==''" />
   </Target>
 
-  <Target Name="PopulateNuspec" BeforeTargets="GenerateNuspec" DependsOnTargets="BuiltProjectOutputGroup;DebugSymbolsProjectOutputGroup;_BuildDependencyProjects">
+  <Target Name="PopulateNuspec" BeforeTargets="GenerateNuspec" DependsOnTargets="BuiltProjectOutputGroup;DebugSymbolsProjectOutputGroup">
 
     <PropertyGroup>
       <!-- Make sure we create a symbols.nupkg. -->

--- a/src/Razor/Razor.Design/test/IntegrationTests/Microsoft.AspNetCore.Razor.Design.Test.csproj
+++ b/src/Razor/Razor.Design/test/IntegrationTests/Microsoft.AspNetCore.Razor.Design.Test.csproj
@@ -35,6 +35,27 @@
     <Reference Include="Microsoft.AspNetCore.Razor.Test.MvcShim.ClassLib" />
   </ItemGroup>
 
+  <Target Name="GenerateBuildVariablesFile" Condition="'$(MSBuildRuntimeType)' == 'Full'" BeforeTargets="BeforeBuild">
+    <PropertyGroup>
+      <GeneratedFileContents>
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
+{
+    internal static partial class BuildVariables
+    {
+        static partial void InitializeVariables()
+        {
+            _msBuildPath = @"$(MSBuildBinPath)\MSBuild.exe"%3B
+        }
+    }
+}
+      </GeneratedFileContents>
+    </PropertyGroup>
+    <WriteLinesToFile Lines="$(GeneratedFileContents)" File="$(BuildVariablesGeneratedFile)" Overwrite="true" WriteOnlyWhenDifferent="true" />
+  </Target>
+  
   <Target Name="EnsureBuildVariablesGeneratedFile">
     <Error Text="BuildVariables.generated.cs was not found. Run .\build /t:Prepare from the root of the repository to generate it." Condition="!Exists('$(BuildVariablesGeneratedFile)')" />
   </Target>

--- a/src/Razor/Razor.Design/test/IntegrationTests/Microsoft.AspNetCore.Razor.Design.Test.csproj
+++ b/src/Razor/Razor.Design/test/IntegrationTests/Microsoft.AspNetCore.Razor.Design.Test.csproj
@@ -13,7 +13,6 @@
     <!-- Copy references locally so that we can use them in the test. -->
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
     <BuildVariablesGeneratedFile>$(MSBuildProjectDirectory)\obj\BuildVariables.generated.cs</BuildVariablesGeneratedFile>
-    <CompileDependsOn>EnsureBuildVariablesGeneratedFile;$(CompileDependsOn)</CompileDependsOn>
   </PropertyGroup>
 
   <ItemGroup>
@@ -35,7 +34,10 @@
     <Reference Include="Microsoft.AspNetCore.Razor.Test.MvcShim.ClassLib" />
   </ItemGroup>
 
-  <Target Name="GenerateBuildVariablesFile" Condition="'$(MSBuildRuntimeType)' == 'Full'" BeforeTargets="BeforeBuild">
+  <Target Name="GenerateBuildVariablesFile" BeforeTargets="BeforeBuild">
+    <Warning Text="Some SDK tests on Windows require the project to be built once using Desktop MSBuild, but the current build is executed using .NET Core MSBuild. This may result in test failures."
+        Condition="!Exists('$(BuildVariablesGeneratedFile)') and '$(MSBuildRuntimeType)' != 'Full' and '$(OS)' == 'Windows_NT'" />
+
     <PropertyGroup>
       <GeneratedFileContents>
 // Copyright (c) .NET Foundation. All rights reserved.
@@ -53,11 +55,7 @@ namespace Microsoft.AspNetCore.Razor.Design.IntegrationTests
 }
       </GeneratedFileContents>
     </PropertyGroup>
-    <WriteLinesToFile Lines="$(GeneratedFileContents)" File="$(BuildVariablesGeneratedFile)" Overwrite="true" WriteOnlyWhenDifferent="true" />
-  </Target>
-  
-  <Target Name="EnsureBuildVariablesGeneratedFile">
-    <Error Text="BuildVariables.generated.cs was not found. Run .\build /t:Prepare from the root of the repository to generate it." Condition="!Exists('$(BuildVariablesGeneratedFile)')" />
+    <WriteLinesToFile Lines="$(GeneratedFileContents)" File="$(BuildVariablesGeneratedFile)" Overwrite="true" WriteOnlyWhenDifferent="true" Condition="'$(MSBuildRuntimeType)' == 'Full'" />
   </Target>
 
 </Project>

